Detectors/TOF/calibration/src/TOFChannelCalibrator.cxx

```diff
@@ -0,0 +1,521 @@
+      LOG(INFO) << "x = " << x.get() << " c " << cnt;
+    }
+  }
+  LOG(INFO) << cnt << " bins inspected";
+}
+
+//_____________________________________________
+void TOFChannelData::printEntries() const
+{
+  // to print number of entries per channel
+  for (int i = 0; i < mEntries.size(); ++i)
+    LOG(INFO) << "channel " << i << " has " << mEntries[i] << " entries";
+}
+
+//_____________________________________________
+int TOFChannelData::findBin(float v) const
+{
+  // find the bin along the x-axis (with t-texp) where the value "v" is; this does not depend on the channel
+  // (axis 1), nor on the sector, so we use sector0
+
+  if (v == mRange)
+    v -= 1.e-1;
+
+  /*
+  LOG(INFO) << "In FindBin, v = : " << v;
+  LOG(INFO) << "bin0 limits: lower = " << mHisto[0].axis(0).bin(0).lower() << ", upper = " << mHisto[0].axis(0).bin(0).upper();
+  LOG(INFO) << "bin1000 limits: lower = " << mHisto[0].axis(0).bin(mNBins-1).lower() << ", upper = " << mHisto[0].axis(0).bin(mNBins-1).upper();
+  LOG(INFO) << "v = " << v << " is in bin " << mHisto[0].axis(0).index(v);
+  */
+  return mHisto[0].axis(0).index(v);
+}
+
+//_____________________________________________
+float TOFChannelData::integral(int chmin, int chmax, float binmin, float binmax) const
+{
+  // calculates the integral in [chmin, chmax] and in [binmin, binmax]
+
+  if (binmin < -mRange || binmax > mRange || chmin < 0 || chmax >= o2::tof::Geo::NCHANNELS)
+    throw std::runtime_error("Check your bins, we cannot calculate the integrals in under/overflows bins");
+  if (binmax < binmin || chmax < chmin)
+    throw std::runtime_error("Check your bin limits!");
+
+  int sector = chmin / o2::tof::Geo::NPADSXSECTOR;
+  if (sector != chmax / o2::tof::Geo::NPADSXSECTOR)
+    throw std::runtime_error("We cannot integrate over channels that belong to different sectors");
+
+  int chinsectormin = chmin % o2::tof::Geo::NPADSXSECTOR;
+  int chinsectormax = chmax % o2::tof::Geo::NPADSXSECTOR;
+  //LOG(INFO) << "Calculating integral for channel " << ich << " which is in sector " << sector
+  //	    << " (channel in sector is " << chinsector << ")";;
+  //  LOG(INFO) << "Bin min = " << binmin << ", binmax = " << binmax <<
+  //             ", chmin = " << chmin << ", chmax = " << chmax <<
+  //             ", chinsectormin = " << chinsectormin << ", chinsector max = " << chinsectormax;
+
+  float res2 = 0;
+  TStopwatch t3;
+  int ind = -1;
+  int binxmin = findBin(binmin);
+  int binxmax = findBin(binmax);
+  LOG(DEBUG) << "binxmin = " << binxmin << ", binxmax = " << binxmax;
+  t3.Start();
+  for (unsigned j = chinsectormin; j <= chinsectormax; ++j) {
+    for (unsigned i = binxmin; i <= binxmax; ++i) {
+      const auto& v = mHisto[sector].at(i, j);
+      res2 += v;
+    }
+  }
+  t3.Stop();
+  LOG(INFO) << "Time for integral looping over axis (result = " << res2 << "):";
+  t3.Print();
+
+  return res2;
+
+  /* // what is below is only for alternative methods which all proved to be slower
```

Comment:
if you are sure this will be never used, better to suppress it.

Reply:
I would prefer to keep it if it is not against the rules.